Add dynamic announcements

jirassimok opened the pull request.

Commit:
Add announcement package to allow dynamic announcements

Commit:
Add "is configured coordinator" as a dynamic announcement property

cawallin reviewed (changes requested):
Nice job getting dynamic announcements working!

jirassimok commented on presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/DiscoveryAgentMap.java:
```diff
@@ -122,7 +121,7 @@ private synchronized void refreshNodes()
                                 properties.get("http"), id), e);
             }
-            Agent agent = new Agent(uri, isCoordinator, isWorker, id);
+            Agent agent = new Agent(uri, isCoordinator, id);
```
That's not currently included in the dynamic announcement, but shouldn't be too hard to add. Until then, I'd rather remove it, so it doesn't _look_ like we support that.

jirassimok commented on presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/announcement/DynamicAnnouncementBinder.java:
```diff
@@ -0,0 +1,230 @@
+ * <p>
+ * Acts as an extension to the {@link Binder} EDSL.
+ * <pre>{@code
+ * dynamicAnnouncementBinder(binder)
```
All right.

jirassimok replied:
All right.

Commit:
Use String for node ID instead of UUID

This change is needed because Airlift's ServiceDescriptor (the basis
for the discovery announcements) uses a String node ID.

jirassimok commented:
I've fixed the typo that was causing the dynamic announcement to be created incorrectly (the property name was wrong).

I've also added a new commit making the Controller's methods use `String` instead of `UUID` for node IDs, because Airlift's `ServiceDescriptor` (the basis for the discovery announcements) uses `String`.

This should be tested on multiple Agents and with proper configuration before it is merged, but I believe it should work.

cawallin commented:
Merged, thanks!
